docs/iris/src/whatsnew/aggregate_directory.py

```diff
@@ -176,9 +260,16 @@ def run_compilation(release):
         for string_line in compiled_text:
             output_object.write(string_line)
 
+
 if __name__ == '__main__':
     PARSER = argparse.ArgumentParser()
+    _DUMMY_VERSION = '9999.9999'
     PARSER.add_argument("release", help="Release number to be compiled",
+                        nargs='?', default=_DUMMY_VERSION,
                         type=version.StrictVersion)
     ARGUMENTS = PARSER.parse_args()
-    run_compilation(ARGUMENTS.release)
+    release = ARGUMENTS.release
+    if release == _DUMMY_VERSION:
+        release = get_latest_release()
+    print('Building release document for release "{!s}"'.format(release))
```

Comment:
Given you've made the `release` argument for `run_compilation()` optional it would be simpler to move this print-out inside that routine. Then you could make the `release` argument of `find_release_directory` mandatory, which "tightens up" your API and makes it less likely to be used incorrectly in future due to a missing argument. It also reduces redundant behaviour and improves the single-point-of-responsibility - i.e. figure out which release you want in one place, then re-use that information.